spec/support/shared_sites_controller_shared_behavior.rb

```diff
@@ -32,7 +32,7 @@
 
   before do
     UserSession.create current_user
-    expect(User).to receive(:find_by_id).and_return(current_user)
```

Comment:
The expectations in `before` blocks were causing some flakey spec behavior in Rails 7. `before` blocks shouldn't include expectations, so I've changed them all to `allow` and cleaned up some other Rubocop offenses.
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecexpectinhook